CRM/Extension/Mapper.php

```diff
@@ -292,7 +292,8 @@ public function getActiveModuleFiles($fresh = FALSE) {
         SELECT full_name, file
         FROM civicrm_extension
         WHERE is_active = 1
-        AND type = "module"
+          AND type = "module"
+        ORDER BY id
```

Comment:
It's really not a good choice between status-quo, `id`, and `file`. :(

What about this policy as a way to transition more gently:

 * On new installations, default to file order -- because that's really the most supportable (i.e. it's consistent across different sites)
 * On existing installations, it continues to work as before -- but you're encouraged to switch the file order

The policy could be implemented with steps like:

1.  in `civicrm.settings.php.template`, add a line like:

   ```php
   define('CIVICRM_DEFAULT_HOOK_ORDER', 'file');
   ```

2. in `CRM/Utils/Check/*`, add a test: `if CIVICRM_DEFAULT_HOOK_ORDER is not defined, display a warning`

3. in `CRM/Extension/Mapper.php`, consult `CIVICRM_DEFAULT_HOOK_ORDER` to decide the actual order (undefined => existing behavior; `file` =>  `ORDER BY file`; `id` =>  `ORDER BY id`)